src/battle-dex.ts

```diff
@@ -492,12 +492,19 @@ const Dex = new class implements ModdedDex {
 				options.shiny = pokemon.shiny;
 				options.gender = pokemon.gender;
 			}
-			if (pokemon.volatiles.dynamax && options.dynamax !== false) isDynamax = true;
-			pokemon = pokemon.getSpeciesForme();
+			let isGigantamax = false;
+			if (pokemon.volatiles.dynamax) {
+				if (pokemon.volatiles.dynamax[1]) {
+					// Gmax sprites are already extremely large, so we don't need to double.
+					isDynamax = false;
+					isGigantamax = true;
+				} else if (options.dynamax !== false) {
+					isDynamax = true;
+				}
+			}
+			pokemon = pokemon.getSpeciesForme() + (isGigantamax ? '-Gmax' : '');
 		}
 		const species = Dex.species.get(pokemon);
-		// Gmax sprites are already extremely large, so we don't need to double.
-		if (species.name.endsWith('-Gmax')) isDynamax = false;
```

Comment:
This code should be left in to retain support for the old system, right? Otherwise, everything else looks good.

Reply:
oh right I didn't think about that